### PR TITLE
transforms: (realize-memref-casts) make compatible with streaming regions

### PR DESCRIPTION
**Pull request by jorendumoulin**
only change needed is to add support for detecting inputs/outputs of streaming regions

### CODE REVIEW

File: compiler/transforms/realize_memref_casts.py [hunk 1]
```diff
@@ -94,11 +95,13 @@ def match_and_rewrite(
                 continue
             # check if input
             is_input = False
-            if not isinstance(use_op, linalg.Generic):
+            if isinstance(use_op, linalg.Generic):
```
Comment (JosseVanDelm) [inline compiler/transforms/realize_memref_casts.py:98]:
```suggestion
            if isinstance(use_op, linalg.Generic) or isinstance(use_op, stream.StreamingRegionOp):
```
Maybe?

File: compiler/transforms/realize_memref_casts.py [hunk 2]
```diff
@@ -114,6 +117,8 @@ def match_and_rewrite(
             is_output = False
             if isinstance(use_op, linalg.Generic):
                 is_output = op.results[0] in use_op.outputs
```
Comment (JosseVanDelm) [inline compiler/transforms/realize_memref_casts.py:119]:
```suggestion
            if isinstance(use_op, linalg.Generic) or isinstance(use_op, stream.StreamingRegionOp):
                is_output = op.results[0] in use_op.outputs
```

**Review by JosseVanDelm — APPROVED** [review]
Looks good! Not sure if the suggestions are too important :)

File: compiler/transforms/realize_memref_casts.py (hunk 1 above)
Comment (jorendumoulin) [reply compiler/transforms/realize_memref_casts.py:98]: pyright does not agree

**Pull request merged**